app/src/crypto.c

```diff
@@ -62,22 +62,23 @@ __Z_INLINE cx_curve_t get_cx_curve(const uint32_t path[HDPATH_LEN_DEFAULT]) {
     }
 }
 
-zxerr_t crypto_extractPublicKey(const flow_path_t *path, uint8_t *pubKey, uint16_t pubKeyLen) {
+zxerr_t crypto_extractPublicKey(const uint32_t path[HDPATH_LEN_DEFAULT], uint8_t *pubKey, uint16_t pubKeyLen) {
     zemu_log_stack("crypto_extractPublicKey");
     MEMZERO(pubKey, pubKeyLen);
 
-    cx_curve_t curve = get_cx_curve(path->data);
+    cx_curve_t curve = get_cx_curve(path);
     if (curve!=CX_CURVE_SECP256K1 && curve!=CX_CURVE_SECP256R1 ) {
         zemu_log_stack("extractPublicKey: invalid_crypto_settings");
         return zxerr_invalid_crypto_settings;
     }
 
-    if (pubKeyLen < PUBLIC_KEY_LEN) {
+    const uint32_t domainSize = 32;
+    const uint32_t pkSize = 1 + 2 * domainSize;
+    if (pubKeyLen < pkSize) {
```

Comment:
we could keep using `PUBLIC_KEY_LEN` that is equal to `65`
```suggestion
    if (pubKeyLen < PUBLIC_KEY_LEN) {
```